WindowsAzureMessaging/WindowsAzureMessaging/Internal/ANHAsync.h

```diff
@@ -0,0 +1,12 @@
+//----------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//----------------------------------------------------------------
+
+#import <Foundation/Foundation.h>
+
+#if __has_attribute(swift_async)
+// Switch to NS_SWIFT_DISABLE_ASYNC once Xcode 13 is available
```

Comment:
So - to make sure I understand:

Apple has introduced a breaking change regarding Swift and async code. Using the flag above, we can toggle between two states: able to compile with XCode 13+ and able to compile with Xcode 12 and below. Correct?

Reply:
Correct, as per the following from the [Xcode 13 Beta release notes](https://developer.apple.com/documentation/xcode-release-notes/xcode-13-beta-release-notes):

> To support the new Swift concurrency model, clang can now warn if you call a completion handler more than once or if an execution path doesn’t have a completion handler call. To turn on the new warning, in your project’s Build Settings in the Apple Clang - Warnings - All languages section, select Yes from the Completion Handler Misuse pop-up menu; alternatively, when invoking clang from the command line, pass the -Wcompletion-handler flag. You can suppress the warning for a method by adding the NS_SWIFT_DISABLE_ASYNC attribute to the method, which also suppresses the translation of this method into an async method in Swift. (10708075)

This attribute is not available on previous versions of Xcode, so in order to support this, we need to define it if it doesn't exist.